src/js/game/hud/parts/building_placer_logic.js

```diff
@@ -114,6 +120,34 @@ export class HUDBuildingPlacerLogic extends BaseHUDPart {
         this.root.camera.upPostHandler.add(this.onMouseUp, this);
     }
 
+    /**
+     * Returns the current base rotation for the current meta-building.
+     * @returns {number}
+     */
+    get currentBaseRotation() {
+        const rotationByBuilding = this.root.app.settings.getAllSettings().rotationByBuilding;
+        if (!rotationByBuilding) {
+            return this.currentBaseRotationGeneral;
+        }
+        const id = this.currentMetaBuilding.get().getId();
```

Comment:
.get() can be null, in which case this would throw an error

Reply:
I added a check for == undefined before calling getId(), defaulting to currentBaseRotationGeneral when currentMetaBuilding isn't set.